.github/workflows/main_ci.yml

```diff
@@ -94,7 +95,7 @@ jobs:
     - name: Configure
       shell: bash
       working-directory: build/
-      run: cmake $GITHUB_WORKSPACE -G"Visual Studio 16 2019" -T ${{matrix.toolset}}
+      run: cmake $GITHUB_WORKSPACE -G"Visual Studio 17 2022" -T ${{matrix.toolset}} -A ${{matrix.arch}}
```

Comment:
Updated this as C++ 20 wasn't being used with the old Clang version in VS 2019, and also to add the v143 toolset to the matrix.